src/Xamarin.Android.Build.Utilities/Sdks/MonoDroidSdkUnix.cs

```diff
@@ -22,6 +22,9 @@ class MonoDroidSdkUnix : MonoDroidSdkBase
 		protected override string FindRuntime ()
 		{
 			string monoAndroidPath = Environment.GetEnvironmentVariable ("MONO_ANDROID_PATH");
+			if (monoAndroidPath == null)
+				// for Mono.Posix and Mono.Data.Sqlite builds in xamarin-android.
+				monoAndroidPath = Path.GetFullPath (Path.Combine (new Uri (GetType ().Assembly.CodeBase).LocalPath, "..", "..", "..", "..", ".."));
```

Comment:
I don't think that is actually better, because in `xabuild` script we set MONO_ANDROID_PATH and in that case my code will never hit that workaround path, while your suggestion will.